yarn-project/acir-simulator/src/avm/opcodes/memory.ts

```diff
@@ -22,8 +22,12 @@ export class Set extends Instruction {
   }
 
   async execute(machineState: AvmMachineState, _journal: AvmJournal): Promise<void> {
-    const res = TaggedMemory.integralFromTag(this.value, this.inTag);
+    // Per the YP, the tag cannot be a field.
+    if ([TypeTag.FIELD, TypeTag.UNINITIALIZED, TypeTag.INVALID].includes(this.inTag)) {
+      throw new InstructionExecutionError(`Invalid tag ${TypeTag[this.inTag]} for SET.`);
+    }
 
+    const res = TaggedMemory.integralFromTag(this.value, this.inTag);
```

Comment:
Naming of routine "integralFromTag" gave me the impression that we cast a tag into an int. Maybe renaming like "integralBaseOnTag" would be a bit more clear.